test/solidity/Facets/GlacisFacet.t.sol

```diff
@@ -0,0 +1,384 @@
+            bridgeData.transactionId,
+            address(uniswap),
+            ADDRESS_DAI,
+            ADDRESS_SRC_TOKEN,
+            swapData[0].fromAmount,
+            bridgeData.minAmount,
+            block.timestamp
+        );
+
+        vm.expectEmit(true, true, true, true, _facetTestContractAddress);
+        emit LiFiTransferStarted(bridgeData);
+
+        initiateSwapAndBridgeTxWithFacet(false);
+
+        // check balances after call
+        assertEq(
+            dai.balanceOf(USER_SENDER),
+            initialDAIBalance - swapData[0].fromAmount
+        );
+        assertEq(USER_SENDER.balance, initialETHBalance - addToMessageValue);
+    }
+
+    function testBase_Revert_BridgeAndSwapWithInvalidReceiverAddress()
+        public
+        virtual
+        override
+    {
+        vm.startPrank(USER_SENDER);
+        // prepare bridgeData
+        bridgeData.receiver = address(0);
+        bridgeData.hasSourceSwaps = true;
+
+        setDefaultSwapDataSingleDAItoSourceToken();
+
+        vm.expectRevert(InvalidReceiver.selector);
+
+        initiateSwapAndBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function testBase_Revert_SwapAndBridgeWithInvalidAmount()
+        public
+        virtual
+        override
+    {
+        vm.startPrank(USER_SENDER);
+        // prepare bridgeData
+        bridgeData.hasSourceSwaps = true;
+        bridgeData.minAmount = 0;
+
+        setDefaultSwapDataSingleDAItoSourceToken();
+
+        vm.expectRevert(InvalidAmount.selector);
+
+        initiateSwapAndBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function testBase_Revert_SwapAndBridgeToSameChainId()
+        public
+        virtual
+        override
+    {
+        vm.startPrank(USER_SENDER);
+        // prepare bridgeData
+        bridgeData.destinationChainId = block.chainid;
+        bridgeData.hasSourceSwaps = true;
+
+        setDefaultSwapDataSingleDAItoSourceToken();
+        dai.approve(_facetTestContractAddress, swapData[0].fromAmount);
+
+        vm.expectRevert(CannotBridgeToSameNetwork.selector);
+
+        initiateSwapAndBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+
+    function testBase_Revert_CallerHasInsufficientFunds()
+        public
+        virtual
+        override
+    {
+        vm.startPrank(USER_SENDER);
+
+        srcToken.approve(
+            address(_facetTestContractAddress),
+            defaultSrcTokenAmount
+        );
+
+        // send all available source token balance to different account to ensure sending wallet has no source token funds
+        srcToken.transfer(USER_RECEIVER, srcToken.balanceOf(USER_SENDER));
+
+        vm.expectRevert(
+            abi.encodeWithSelector(
+                InsufficientBalance.selector,
+                bridgeData.minAmount,
+                0
+            )
+        );
+
+        initiateBridgeTxWithFacet(false);
+        vm.stopPrank();
+    }
+}
+
+contract GlacisFacetWormholeTest is GlacisFacetTestBase {
```

Comment:
I feel the ordering could be improved as these two smaller contracts with their setup functions are more closely connected to the setup function above. How do you feel about that?

Reply:
hmm, tbh I don't know what can be improved. The base test contract shares setup function with all contracts that inherits from it and children setup only necessery parameters. 

Do you have something in mind how I could improve it?

One drawback is that when opening the file you need to scroll all the way to the bottom to see the testing parameters. There’s a trade-off between “top-down” logical ordering and quick visibility.

Reply:
I personally would move these two contracts to the beginning of the file.
Or have it all integrated in one test contract like we do for other facets. 

The drawback you mentioned is what triggered me to write this comment.

Reply:
thanks for suggestion. The thing is that Solidity enforces that child contracts must be defined after parent... If I would move them it won't compile. Merging everything into one contract would reduce clarity that abtraction gives us. It's nice to keep tests modular.

Third approach would be to have seperated folder for `GlacisFacet` with `GlacisFacetTestSuite.t.sol`, `GlacisFacetWormhole.t.sol`, `GlacisFacetLink.t.sol` but IMO  it's overkill.

I would like to have opinion from @ezynda3 as well.

Reply:
I wasnt aware of this Solidity limitation. 
No, then feel free to leave it as-is. I dont think it makes sense to split into separate files.